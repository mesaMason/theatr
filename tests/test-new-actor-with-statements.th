dolphin(int weight, int age):
    print("hello from inside dolphin")
    print(weight);
    print(age)
    int sum = weight + age
    print(sum)
    receive:
        eat(int food):
            weight = weight + food
    drop:
        print("inside drop()")
    after:
        return;
        
func main() -> int:
    int weight;
    int age;
    weight = 10;
    age = 3;
    actor d
    d = new dolphin(weight, age)    
    die() | d
    return 0

