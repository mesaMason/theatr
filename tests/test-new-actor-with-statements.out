new dolphin created
hello from inside dolphin
44
60
104
hello from inside dolphin
10
3
13